### ISSUE TITLE
Distribution in FIM calculation 

### ISSUE THREAD

**Issue by lan-qing** [tudor-berariu/fisher-information-matrix#2]
Hi Tudor,

I feel grateful that you release the codes and they are helpful. However, I have a question here.

L48 - L52 in fim.py:

    logits = model(data)        
    if empirical:           
        outdx = target.unsqueeze(1)        
    else:            
        outdx = Categorical(logits=logits).sample().unsqueeze(1).detach()

Here the logits are the output of the model, whose last layer is `log_softmax`. So the logits might be negative rather than rational probabilities, which makes `outdx` meaningless. Am I understanding wrong somewhere? Or should we add a line `logits = torch.exp(logits)`?

Thanks for your attention.
